akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -50,9 +62,33 @@ abstract class GraphStageWithMaterializedValue[+S <: Shape, +M] extends Graph[S,
 }
 
 /**
- * A GraphStage represents a reusable graph stream processing stage. A GraphStage consists of a [[Shape]] which describes
- * its input and output ports and a factory function that creates a [[GraphStageLogic]] which implements the processing
- * logic that ties the ports together.
+ * Java API: A GraphStage represents a reusable graph stream processing stage.
+ *
+ * Extend this `GraphStageWithMaterializedValue` if you want to provide a materialized value,
+ * represented by the type parameter `M`. If your GraphStage does not need to provide a materialized
+ * value you can instead extende [[GraphStage]] which materializes a [[NotUsed]] value.
```

Comment:
AbstractGraphStage?

Reply:
Thanks

Reply:
Oh, actually no; we only have one type there :) the handlers have abstract versions

Reply:
(`abstract` graphstage versions are only needed if we return a pair/tuple; in normal case it's just the logic)